### PR TITLE
Pia 4595 extract feedback from cleanup

### PR DESCRIPTION
**Pull request by abolfazlimahdi**

### CODE REVIEW

File: capture-sdk/sdk/src/main/java/net/gini/android/capture/GiniCapture.java [hunk 1]
```diff
@@ -154,123 +159,139 @@ public static synchronized boolean hasInstance() {
...
+     * @param amount           accepts extracted amount and currency
      */
-    public static synchronized void cleanup(@NonNull final Context context,
-                                            @NonNull final String paymentRecipient,
-                                            @NonNull final String paymentReference,
-                                            @NonNull final String paymentPurpose,
-                                            @NonNull final String iban,
-                                            @NonNull final String bic,
-                                            @NonNull final Amount amount) {
+    public static synchronized void transferSummary(@NonNull final String paymentRecipient, @NonNull final String paymentReference, @NonNull final String paymentPurpose, @NonNull final String iban, @NonNull final String bic, @NonNull final Amount amount) {
```
Comment (a-szotyori) [inline capture-sdk/sdk/src/main/java/net/gini/android/capture/GiniCapture.java:212]: [Request] We should start the method name with a verb to make it clearer what the method does:
```suggestion
    public static synchronized void sendTransferSummary(@NonNull final String paymentRecipient, @NonNull final String paymentReference, @NonNull final String paymentPurpose, @NonNull final String iban, @NonNull final String bic, @NonNull final Amount amount) {
```

File: bank-sdk/example-app/src/main/java/net/gini/android/bank/sdk/exampleapp/ui/ExtractionsActivity.kt [hunk 2]
```diff
@@ -98,23 +106,29 @@ class ExtractionsActivity : AppCompatActivity(), ExtractionsAdapter.ExtractionsA
...
             }
         }
     }
 
-    private fun <T> getSortedExtractions(extractions: Map<String, T>): List<T> = extractions.toSortedMap().values.toList()
+    private fun <T> getSortedExtractions(extractions: Map<String, T>): List<T> =
+        extractions.toSortedMap().values.toList()
 
     private fun sendFeedbackAndClose(binding: ActivityExtractionsBinding) {
         // Feedback should be sent only for the user visible fields. Non-visible fields should be filtered out.
```
Comment (a-szotyori) [inline bank-sdk/example-app/src/main/java/net/gini/android/bank/sdk/exampleapp/ui/ExtractionsActivity.kt:134]: [Request] Rename to feedback summary:
```suggestion
    private fun sendTransferSummaryAndClose(binding: ActivityExtractionsBinding) {
        // Transfer summary should be sent only for the user visible fields. Non-visible fields should be filtered out.
```

File: capture-sdk/sdk/src/main/java/net/gini/android/capture/GiniCapture.java [hunk 3]
```diff
@@ -154,123 +159,139 @@ public static synchronized boolean hasInstance() {
...
+            cleanup(context);
         }
         return new Builder();
     }
+
     /**
-     * Destroys the {@link GiniCapture} instance and frees up used resources.
+     * Provides transfer summary to Gini.
      *
      * <p>Please provide the required extraction feedback to improve the future extraction accuracy.
```
Comment (a-szotyori) [inline capture-sdk/sdk/src/main/java/net/gini/android/capture/GiniCapture.java:196]: [Request] Rename to transfer summary.
```suggestion
     * <p>Please provide the required transfer summary to improve the future extraction accuracy.
```

**Review by a-szotyori — CHANGES_REQUESTED** [review]
Sorry for the many requests. Most of them are about the documentation.

Please also update the extraction feedback integration tests:
* Rename to `TransferSummaryIntegrationTest`
* Replace deprecated `cleanup()` with `sendTransferSummary()` and `cleanup()`.

The integration tests can be found here:
* capture-sdk/default-network/src/androidTest/java/net/gini/android/capture/network/ExtractionFeedbackIntegrationTest.kt
* bank-sdk/sdk/src/androidTest/java/net/gini/android/bank/sdk/ExtractionFeedbackIntegrationTest.kt

Thank you!

File: capture-sdk/sdk/src/main/java/net/gini/android/capture/GiniCapture.java [hunk 4]
```diff
@@ -71,14 +70,20 @@
...
- * while also providing the required extraction feedback to improve the future extraction accuracy.
+ * <p> Use the {@link #cleanup(Context)} method to clean up the SDK
+ * Please follow the recommendations below:
+ *
+ * <ul>
+ *     <li>Do cleanup after TAN verification.to clean up.</li>
+ * </ul>
+ *
+ * <p> Use the {@link #transferSummary(String, String, String, String, String, Amount)}  method to
+ * provide the required extraction feedback to improve the future extraction accuracy.
```
[the remark this answers is not in the document]
Comment (abolfazlimahdi) [reply capture-sdk/sdk/src/main/java/net/gini/android/capture/GiniCapture.java:81]: Done with a modification.

File: bank-sdk/sdk/src/main/java/net/gini/android/bank/sdk/GiniBank.kt [hunk 5]
```diff
@@ -127,12 +130,44 @@ object GiniBank {
...
+     *
+     *
+     * @param paymentRecipient payment receiver
+     * @param paymentReference ID based on Client ID (Kundennummer) and invoice ID (Rechnungsnummer)
+     * @param paymentPurpose statement what this payment is for
+     * @param iban international bank account
+     * @param bic bank identification code
+     * @param amount accepts extracted amount and currency
+     */
+    fun transferSummary(
```
[the remark this answers is not in the document]
Comment (abolfazlimahdi) [reply bank-sdk/sdk/src/main/java/net/gini/android/bank/sdk/GiniBank.kt:158]: Oh my bad, I thought the naming is in the ticket title so I should not change it.

File: capture-sdk/sdk/src/main/java/net/gini/android/capture/GiniCapture.java [hunk 6]
```diff
@@ -154,123 +159,139 @@ public static synchronized boolean hasInstance() {
...
+     * </ul>
+     *
+     * @param context          Android context
+     * @param paymentRecipient payment receiver
+     * @param paymentReference ID based on Client ID (Kundennummer) and invoice ID (Rechnungsnummer)
+     * @param paymentPurpose   statement what this payment is for
+     * @param iban             international bank account
+     * @param bic              bank identification code
+     * @param amount           accepts extracted amount and currency
+     * @deprecated Please use {@link #cleanup(Context)} to cleanup and {@link #transferSummary(String, String, String, String, String, Amount)} to provide the required extraction feedback.
```
[the remark this answers is not in the document]
Comment (a-szotyori) [reply capture-sdk/sdk/src/main/java/net/gini/android/capture/GiniCapture.java]: @mahdiabolfazli I think you missed this one.

File: bank-sdk/sdk/src/main/java/net/gini/android/bank/sdk/GiniBank.kt [hunk 7]
```diff
@@ -150,7 +185,13 @@ object GiniBank {
      * @param iban international bank account
      * @param bic bank identification code
      * @param amount accepts extracted amount and currency
+     *
+     * @deprecated Please use {@link #releaseCapture(Context)} which does not require transfer summary parameters.
      */
+    @Deprecated(
+        "Please use releaseCapture(Context) which does not require transfer summary parameters.",
+        ReplaceWith("releaseCapture(context)")
+    )
```
[the remark this answers is not in the document]
Comment (a-szotyori) [reply bank-sdk/sdk/src/main/java/net/gini/android/bank/sdk/GiniBank.kt:194]: @mahdiabolfazli I think you missed this one.

File: bank-sdk/sdk/src/main/java/net/gini/android/bank/sdk/GiniBank.kt [hunk 8]
```diff
@@ -127,12 +130,44 @@ object GiniBank {
...
+     *  Provide the final data approved by the user (and not the initially extracted only).
+     *
+     *
+     * @param paymentRecipient payment receiver
+     * @param paymentReference ID based on Client ID (Kundennummer) and invoice ID (Rechnungsnummer)
+     * @param paymentPurpose statement what this payment is for
+     * @param iban international bank account
+     * @param bic bank identification code
+     * @param amount accepts extracted amount and currency
+     */
```
[the remark this answers is not in the document]
Comment (a-szotyori) [reply bank-sdk/sdk/src/main/java/net/gini/android/bank/sdk/GiniBank.kt]: @mahdiabolfazli I think you missed this one.

**Review by a-szotyori — APPROVED** [review]
Nice! 👏

File: bank-sdk/sdk/src/main/java/net/gini/android/bank/sdk/GiniBank.kt [hunk 9]
```diff
@@ -127,16 +130,46 @@ object GiniBank {
...
-            .build()
+        GiniCapture.newInstance(context).applyConfiguration(captureConfiguration).build()
         giniCapture = GiniCapture.getInstance()
     }
 
+
+    /**
+     * Provides transfer summary to Gini.
+     *
+     * Please provide the required transfer summary to improve the future extraction accuracy.
```
Comment (zladzeyka) [inline bank-sdk/sdk/src/main/java/net/gini/android/bank/sdk/GiniBank.kt:141]: [Suggestion] we have 4 sentences in row, which are starting from `Please`. I propose to keep it till `Please follow the recommendations below` and remove other please after.

Comment (abolfazlimahdi) [reply bank-sdk/sdk/src/main/java/net/gini/android/bank/sdk/GiniBank.kt:141]: @zladzeyka Just to be on the same page, you mean I delete the `please`s after the first one?

**Pull request merged**